File: challenge_3/challenge_3.exs
defmodule PrimeFactor do
	def find_prime(limit) do
		factor(2, 0, limit)
	end

	def factor(counter, largest, limit) when (counter*counter) < limit do
		if rem(limit,counter) == 0 do
			limit = round(limit/counter)
			largest = counter
		else
			counter = counter + 1
		end
		factor(counter, largest, limit)
	end

	def factor(counter, largest, limit) when (counter*counter) >= limit do
		if limit > largest, do: largest = limit
		largest
	end
end

IO.puts PrimeFactor.find_prime(600_851_475_143)

File: challenge_3/challenge_3.c
#include <stdio.h>

int main(void)
{
  int largest;
  int counter = 2;
  long new_num = 600851475143;

  while (counter * counter <= new_num) {
    if (new_num % counter == 0) {
      new_num /= counter;
      largest = counter;
    } else {
      counter += 1;
    }

    if(new_num > largest) largest = new_num;
  }

  printf("%d\n", largest);
  return 0;
}
